email/exporter.go

```diff
@@ -100,14 +102,24 @@ func (impl *ExporterImpl) SendContacts(ctx context.Context, req *emailpb.SendCon
 		return nil, berrors.InternalServerError("Incomplete gRPC request message")
 	}
 
+	var unseen []string
+	for _, email := range req.Emails {
+		// We check the cache here to avoid enqueuing an email that has already
+		// been seen, but we do not add it to the cache until the Pardot client
+		// has successfully sent it.
+		if !impl.emailCache.Seen(email) {
```

Comment:
I think this check should be excluded. It makes sense to avoid enqueueing duplicate addresses, for sure. But it has two downsides, imo:
1) weird abstraction layering, where the ExporterImpl both has-a Cache and has-a PardotClient, and then that PardotClient also has-a Cache and it's the same cache! This triangular dependency feels very weird, and is a pattern I'd usually try to avoid.
2) as mentioned above, duplicate metric increments, since most emails that aren't in the cache will be checked for inclusion twice

I think it's significantly simpler and cleaner for only the PardotClient to be responsible for the cache, and I don't think I'm very concerned about the cost of the larger queue.